server/server/src/cold_chain/sensor.rs

```diff
@@ -0,0 +1,139 @@
+use actix_web::{
+    http::header,
+    web::{self, Data},
+    HttpRequest, HttpResponse,
+};
+use log::error;
+use mime_guess::mime;
+use regex::Regex;
+use repository::{get_sensor_type, RepositoryError};
+use service::{
+    auth_data::AuthData,
+    sensor::{insert::InsertSensor, update::UpdateSensor},
+    service_provider::ServiceProvider,
+};
+
+use super::validate_request;
+
+#[derive(serde::Deserialize)]
+pub struct Sensor {
+    id: String,
+    #[serde(rename = "macAddress")]
+    mac_address: String,
+    #[serde(rename = "logInterval")]
+    log_interval: i32,
+    #[serde(rename = "programmedDate")]
+    programmed_date: usize,
+    name: String,
+    #[serde(rename = "logDelay")]
+    log_delay: Option<i32>,
+    #[serde(rename = "batteryLevel")]
+    battery_level: i32,
+}
+
+pub async fn sensors(
```

Comment:
Could this be renamed to `put_sensors` or `put_sensor` (endpoint is `sensor`, or should the endpoint be `sensors` as well?)

Similar for login, rename it to `post_login` (why is it POST and not GET?)

Open for other naming conventions, but this would make it clear that the methods are the direct handlers for the endpoints...

Reply:
thanks - much better 🙏 

( It's using POST because that's what the cold chain app is doing )